lib/rex/post/mssql/ui/console.rb

```diff
@@ -0,0 +1,147 @@
+# -*- coding: binary -*-
+
+module Rex
+  module Post
+    module MSSQL
+      module Ui
+        ###
+        #
+        # This class provides a shell driven interface to the MSSQL client API.
+        #
+        ###
+        class Console
+          include Rex::Ui::Text::DispatcherShell
+
+          # Dispatchers
+          require 'rex/post/mssql/ui/console/command_dispatcher'
+          require 'rex/post/mssql/ui/console/command_dispatcher/core'
+          require 'rex/post/mssql/ui/console/command_dispatcher/client'
+          require 'rex/post/mssql/ui/console/command_dispatcher/modules'
+
+          #
+          # Initialize the MSSQL console.
+          #
+          # @param [Msf::Sessions::MSSQL] session
+          def initialize(session, opts={})
+            # The mssql client context
+            self.session = session
+            self.client = session.client
+            self.cwd = session.client.mssql_query('SELECT DB_NAME();')[:rows][0][0]
```

Comment:
Not a blocker; Looks like this information is already given to us during the login negotiation process via wireshark:

![image](https://github.com/rapid7/metasploit-framework/assets/60357436/4525efd4-4c8f-4ff7-bb26-858e9eeef579)

And reading the spec:

> The type of environment change:
>
> 1: Database
> https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-tds/2b3eb7e5-d43d-4d1b-bf4d-76b9e3afc791

So to do this without an extra query, all we'd need to do is something similar to using the info that we parse here:

```
From: /Users/user/Documents/code/metasploit-framework/lib/rex/proto/mssql/client.rb:414 Rex::Proto::MSSQL::Client#mssql_login:

    409: 
    410:           info = {:errors => []}
    411:           info = mssql_parse_reply(resp, info)
    412: 
    413:           require 'pry-byebug'; binding.pry
 => 414:           return false if not info
    415:           info[:login_ack] ? true : false
    416:         end
    417: 
    418:         #
    419:         #this method send a prelogin packet and check if encryption is off

[1] pry(#<Rex::Proto::MSSQL::Client>)> info
=> {:errors=>[],
 :envs=>
  [{:type=>1, :old=>"master", :new=>"master"},
   {:type=>7, :old=>"", :new=>"\t\x04\xD04"},
   {:type=>2, :old=>"", :new=>"us_english"},
   {:type=>4, :old=>"4096", :new=>"4096"}],
 :infos=>
  ["SQL Server Info #5701 (State:2 Severity:0): Changed database context to 'master'.",
   "SQL Server Info #5703 (State:1 Severity:0): Changed language setting to us_english."],
 :login_ack=>true,
 :done=>{:status=>0, :cmd=>0, :rows=>0}}
[2] pry(#<Rex::Proto::MSSQL::Client>)> 
```

And pull out the database type using the `type=>1` reference from the docs:

```
  [{:type=>1, :old=>"master", :new=>"master"},
```

And expose it on the mssql client instance for future users